### PR TITLE
add event weights

### PR DESCRIPTION
**Pull request by llayer**
Hi Giles,

I added a first version to include the event weights, however the function to_shape (https://github.com/GilesStrong/pytorch_inferno/blob/master/pytorch_inferno/inferno.py#L87) is defined as static, so I cannot access the model wrapper, but have to pass the weights as argument. Not sure whether this is a good solution like this, it would be great if you could give it a look. 

Also I am not sure whether I have to call .detach() on the weights tensor when calculating the asimov variables, I think it is not necessary since the tensor does not depend on any of the variables that are differentiated.

### CODE REVIEW

**Comment by GilesStrong** [comment]
Thanks Lukas! I think passing the weights to the function is a good idea, since the function isn't aware of whether to filter signal or bkg. I just added the same weight passing to the approximated INFERNO, too, and cleaned the notebooks.
There's a problem with the tests, so we shouldn't expect them to pass.

**Comment by GilesStrong** [comment]
Ah cool, the tests are passing again. Merging!
